openml-h2o/src/test/java/com/feedzai/openml/h2o/H2OModelProviderTrainTest.java

```diff
@@ -228,4 +229,19 @@ public final void testIsolationForestWithNotEnoughInstances() {
                 .doesNotThrowAnyException();
     }
 
+    /**
+     * Tests that the right exception is thrown when the dataset is empty
+     * @since 1.0.9
+     */
+    @Test
+    public final void testExceptionIsThrownWhenDatasetIsEmpty() {
+        final Dataset dataset = createDataset(SCHEMA, 0);
+        final Map<String, String> params = H2OAlgorithmTestParams.getIsolationForest();
+        final H2OModelCreator loader = getMachineLearningModelLoader(H2OAlgorithm.ISOLATION_FOREST);
+        final Random random = new Random(234);
+        final Throwable thrown = catchThrowable(() -> loader.fit(dataset, random, params));
+
+        assertThat(thrown).isInstanceOf(ModelTrainingException.class)
```

Comment:
Instead of using `catchThrowable` + `assertThat` you can simply do `assertThatThrownBy`.

Source: https://joel-costigliola.github.io/assertj/